### ISSUE TITLE
Default shortcuts for mac

### ISSUE THREAD

**Issue by goanpeca** [spyder-ide/spyder#10854]
On a macbook pro some of the default keyboard shortcuts of Spyder collide with "normal" functions on OSX.

Home/End (<kbd>Command+Up</kbd>, <kbd>Command+Down</kbd>) are taken by Move to previous/next cell.

Also, moving to the start and end of a file (<kbd>Command+Left</kbd>, <kbd>Command+Right</kbd>) is taken by move to next/previous word, which on Mac make more sense to use <kbd>Alt+Left</kbd>, <kbd>Alt+Right</kbd>, instead of command.

**Comment by ccordoba12** [comment]
I think this depends on the second set of shortcuts @jnsebgosselin is working on, right?

Or do you prefer to have the shortcuts above as the primary ones in macOS?

**Comment by goanpeca** [comment]
>Or do you prefer to have the shortcuts above as the primary ones in macOS?

I think they should be the primary one, because the point is to have the "normal" mac commands first. Also having an additional set of shortcuts would not help as they are currently colliding with the default Mac actions.

Any thoughts here @impact27 ?

**Comment by goanpeca** [comment]
As in it is very annoying not being able to move to the start or end of file with <kbd>Cmd+Up/Cmd+Down</kbd> as in any other IDE/TexteEditor on mac

**Comment by impact27** [comment]
I agree that we should follow the guidlines for shortcuts (https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/keyboard/) and user expectations as a default. The Windows shortcuts should be an option.

For example, from the guidelines for Up arrow:

Shift-Control-Up arrow | ● | Extend selection upward in the next semantic unit, typically the beginning of the document.
 - Doesn't do anything

Shift-Up arrow | ● | Extend selection to the line above, to the nearest character boundary at the same horizontal location.
 - Works

Option-Shift-Up arrow | ● | Extend selection to the beginning of the current paragraph, then to the beginning of the next paragraph.
 - duplicates the current line

Control-Up arrow | ● | Move focus to another value or cell within a view, such as a table.
 - Moves to the beginning of the document

And not in the guidelines, but present in apple apps such as textedit:
Cmd+Up : Move to the beginning of the document
 - Moves to the previous cell

**Comment by ccordoba12** [comment]
Ok, then I agree with this. @goanpeca, please evaluate if we can do this for 4.1 or we should leave it for 5.

**Comment by anthony-langford** [comment]
Has this been implemented?

**Comment by 7hx1e** [comment]
I am also curious about the progress, as this is the single most reason I switched back to VSCode, which their interactive window I personally dislike.

**Comment by ccordoba12** [comment]
@7hx1e, unfortunately we haven't had time to work on this.

@impact27, any chance you have some free time to help us this one? If not, I'm afraid it'll have to wait until Spyder 6.

**Comment by markmorr** [comment]
Just checking in on this! Has anyone found a fix or workaround? I would just love a way to get Cmd + left/right arrow back to the Mac default behavior. But there doesn't seem to be an option for "jump to beginning/end of line" in the keyboard shortcuts

**Comment by ccordoba12** [comment]
@markmorr, about going to beginning/end of line, there is a regression in Spyder 6 that removed those shortcuts from Preferences. But they'll be restored in 6.0.3, to be released in a couple of weeks, so you'll be able to customize then.

This is a screenshot from PR #23024, which fixes that problem:

![imagen](https://github.com/user-attachments/assets/76739822-238d-4f0b-8e10-83cdb08504da)

**Comment by dongYoun2** [comment]
@ccordoba12 Hi, is command + left/right on Mac restored as moving the cursor to the front/end of the line in version 6.0.2? Thanks

**Comment by ccordoba12** [comment]
@dongYoun2, nop, that's not the case, sorry. What I said is that you can change that in Preferences to make things work as in other Mac apps.
